src/main/java/com/michelin/ns4kafka/controller/acl/AclController.java

```diff
@@ -105,10 +108,25 @@ public HttpResponse<AccessControlEntry> apply(Authentication authentication,
                                                   @QueryValue(defaultValue = "false") boolean dryrun) {
         Namespace ns = getNamespace(namespace);
 
+        boolean grantorIsSecured = getNamespace(accessControlEntry.getMetadata().getNamespace()).getSpec().isSecured();
+        boolean granteeIsPublic = PUBLIC_GRANTED_TO.equals(accessControlEntry.getSpec().getGrantedTo());
+        boolean granteeIsSecured = !granteeIsPublic
```

Comment:
I put this additional condition `!granteeIsPublic`, because when the ACL is granted to `*`, `getNamespace("*")` will fail.

Feel free to challenge if you think there is a prettier way to forbid secured namespaces granting ACL to all or to basic namespaces.